tests.py

```diff
@@ -26,66 +35,135 @@ def test_g1_p1(self):
             self.assertAlmostEqual(v, rdata[k])
 
     def test_g2_p2a(self):
-        expected = {'precision': 0.5, 'recall': 1.0, 'f1': 0.6666666666666666, 'tpr': 1.0, 'fpr': 1.0}
+        expected = {'global_precision': 0.5,
+                    'global_recall': 1.0,
+                    'global_f1': 0.6666666666666666,
+                    'global_tpr': 1.0,
+                    'global_fpr': 1.0,
+                    'average_precision': 0.5,
+                    'average_recall': 1.0,
+                    'average_f1': 0.619047619,
+                    'average_tpr': 1.0,
+                    'average_fpr': 1.0}
         r = evaluation_script.evaluate("testfiles/ground_truth_set2.tsv",
                                        "testfiles/predictions_set2a.tsv", "supervised")
         rdata = r['result'][0]['data']
         for k, v in expected.items():
             self.assertAlmostEqual(v, rdata[k])
 
     def test_g2_p2b(self):
-        expected = {'precision': 1, 'recall': 0.0, 'f1': 0.0, 'tpr': 0.0, 'fpr': 0.0}
+        expected = {'global_precision': 1,
+                    'global_recall': 0.0,
+                    'global_f1': 0.0,
+                    'global_tpr': 0.0,
+                    'global_fpr': 0.0,
+                    'average_precision': 1.0,
+                    'average_recall': 0.142857143,
```

Comment:
Hm... Shows a logical problem with the query metrics. Oh well. It's correct mathematically, not much to be done.

In this case the predictions are saying that nothing is relevant. However, for queries with no relevant results this has a recall at `1.0`, which get averaged in. This could actually occur in the data challenge, as I didn't take any steps to ensure that every query has at least one relevant judgement in the evaluation set, and a few queries have only a small number of relevant results.